### DISCUSSION TITLE
개발하면서 궁금한 점 몇 가지 질문드립니다! :)

### DISCUSSION THREAD

**Discussion by kimyang-sun** [naverz/zepeto-studio-kor#895] [5. Q&A]
안녕하세요! :)
유니티 제페토 스크립트 개발을 하게 되어서 하나하나 차근차근 공부하고 있는데 하다보니 궁금한 점 몇 가지랑 제가 찾아본 내용이 맞는지 궁금증이 생겨서 질문드립니다!

1. 우선 빌드잇에서 작업 파일을 유니티로 불러올 때 상호작용을 제외한 오브젝트만 불러오는거로 알고있습니다!
여기서 상호작용(앉기, 눕기, 자동차 대여, 자동차 타기, 운전) 등은 직접 유니티에서 따로 개발해주어야 하는 부분인거 맞을까요?.?
제공해주는 제스쳐는 패키지 내부에 _Resources>Animations 폴더에 있는 제스처들이 전부인거죠?

2. 유니티 앱에 접속해서 월드에 들어갔을때 나오는 기본적인 UI (사진찍기, 채팅창, 제스쳐 등) 는 제페토에 등록하여 심사를 받고 공식 월드가 되면 기본적으로 제공이 되는 부분일까요??
여기서 심사를 받기 전 단계에서 테스트 링크로 테스트할때는 기본 UI가 없는 상태로 나오는지도 궁금합니다!

3. 제스쳐 애니메이션에 대한 글을 봤습니다! #154 
앉기나 무언가를 타는 제스쳐는 기본 제공 제스쳐에 없는거 같은데 해당 제스쳐들을 구현하려면 3D툴로 직접 애니메이션 anim 파일을 만들어 유니티에 넣어 Rig에서 "휴머노이드" 설정 후 Apply 눌러 생성된 세모난 아이콘의 애니메이션 클립을 이용하는 방법으로 
제스쳐를 구현하면 맞는걸까요??

4. 자동차 기능 [ex. 자동차 타고 운전하는 기능]
빌드잇에서는 자동차 대여 상호작용이 있는 오브젝트를 통해 자동차를 타고 이동하는 기능이 있는데
해당 기능을 유니티 제페토 스크립트로 구현하려면 직접 코드를 개발해주어야 하는게 맞는걸까요?? (예제 같은거도 없겠죠??)

5. UI 중에 상호작용 아이콘 (ex. 의자에 다가갔을때 뜨는 손가락 아이콘, 어떠한 물건에 다가갔을때 톱니바퀴 아이콘) 
이런 아이콘들은 아이콘들을 직접 디자인해서 오브젝트로 넣어주는거일까요?? 월드마다 아이콘들이 다 동일하게 되어있는거 같아서
혹시 이런부분도 어디서 제공을 해주는 게 있나 궁금합니다!

7. 월드맵을 구경하다가 씬이동이 있는 맵에서 포탈을 누르니까 [이동하시겠습니까? 예 아니오] 알림창이 뜨는걸 봤습니다. 
해당 confirm창 UI도 기본으로 제페토에서 제공해주는 스타일이 있는걸까요?? 아니면 오브젝트로 직접 만들어서 띄워주어야 하는걸까요??
그 후에 이동할 때 제페토 자체 로딩창이 나오던데 유니티에서 씬이동 기능을 넣으면 월드로 배포후에는 알아서 제페토 로딩창이 자동으로 나오는 구조일까요??

적다보니 너무 많이 적은거 같은데 읽어주셔서 감사합니다. :)

**Comment by minjujuu** [comment]
1. 가이드를 참고해보면 상호작용 오브젝트는 동작하지 않을 수 있어서 따로 개발해주는 게 맞을 것 같습니다.
- 참고: https://github.com/naverz/zepeto-studio-kor/discussions/509
- 참고 : https://studio.zepeto.me/guides/loading-buildit-maps

2. 채팅창은 기본으로 노출되는 것이 맞고, 사진찍기는 샘플 프로젝트를 이용할 수 있습니다. 
샘플에도 간단한 제스처 기능이 있지만 기본 3종만 있어서 이 외 제스처는 애니메이션을 따로 추가해야 할 것 같습니다.
- 샘플 프로젝트 : https://studio.zepeto.me/kr/guides/screenshot
- 제스처 관련 함수 참고 (SetGesture, CancelGesture) : https://studio.zepeto.me/kr/guides/zepetocharacter

3. 3D 툴로 직접 만들 수도 있고, Mixamo라는 곳에서 무료 애니메이션을 다운받아 활용해볼 수도 있습니다.
그리고 말씀주신대로 휴머노이드 설정 후 생성된 애니메이션 클립을 이용할 수 있습니다. (위의 SetGesture 활용)

4. 자동차 타고 운전하는 기능은 따로 구현해야 할 것 같습니다..!

5. 위 2번에서 말씀드린 샘플 프로젝트를 보시면 몇 가지 제공되는 UI 이미지는 있으나,
그 외에 UI의 경우에는 직접 디자인해서 오브젝트로 넣어주어야 할 것 같습니다.

6. confirm창 UI도 따로 구현해야 할 것 같습니다. 
제페토 자체 로딩창은 처음 월드 진입시에만 뜨고, 씬 이동 기능을 넣을 때 제페토 로딩창이 자동으로 나오는 구조는 아닙니다.

아래와 같은 공식 샘플도 있으니 참고해보시면 좋을 것 같습니다! 
[[Example] 공식 샘플 월드 Official World Samples #615](https://github.com/naverz/zepeto-studio-kor/discussions/615)

**Reply by kimyang-sun** [reply]
빠른 답변 정말 감사합니다!

**Reply by kimyang-sun** [reply]
@minjujuu 
![ui](https://user-images.githubusercontent.com/56675004/174265103-2acba60f-cee0-402f-a79e-5a740c7ce76b.jpg)

아 혹시! 2번 답변에서 채팅창 제외하고 나머지는 샘플을보고 구현해야한다고 하셨는데
첨부 이미지에서 빨간박스 안에 있는 부분들은 모든 맵에서 다 동일하게 구성되어 있는 부분 같은데 이 부분도 다 직접 구현이 필요한걸까요???

**Reply by minjujuu** [reply]
해당 맵은 제페토 스튜디오로 만든 맵이 아닌 것으로 보입니다!
제페토 스튜디오로 만든 월드의 경우 빨간 박스 안에 있는 부분들이 자동으로 적용되지는 않습니다.
